duckshoot.py

```diff
@@ -0,0 +1,337 @@
+        return self._colour
+
+    @colour.setter
+    def colour(self, colour):
+        self._colour = colour
+
+    def run(self):
+        # This method runs in a separate thread
+        while not self.terminated:
+            # Wait for an image to be written to the stream
+            if self.event.wait(1):
+                try:
+                    # Read the image and do some processing on it
+                    self.stream.seek(0)
+                    self.process_image(self.stream.array, self.screen)
+                finally:
+                    # Reset the stream and event
+                    self.stream.seek(0)
+                    self.stream.truncate()
+                    self.event.clear()
+
+    # Image processing function
+    def process_image(self, image, screen):
+        screen = pygame.display.get_surface()
+        # crop image to speed up processing and avoid false positives
+        image = image[self.CROP_V_OFFSET:(self.CROP_V_OFFSET+self.CROP_HEIGHT), self.CROP_H_OFFSET:(self.CROP_H_OFFSET+self.CROP_WIDTH)]
+        img = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+        if not self.menu:
+            frame = pygame.surfarray.make_surface(cv2.flip(img, 1))
+            screen.fill([0, 0, 0])
+            font = pygame.font.Font(None, 24)
+            screen.blit(frame, (0, 0))
+        image = cv2.medianBlur(image, 11)
+        # Convert the image from 'BGR' to HSV colour space
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2HSV)
+        # We want to extract the 'Hue', or colour, from the image. The 'inRange'
+        # method will extract the colour we are interested in (between 0 and 180)
+        default_colour_bounds = ((40, 0, 0), (180, 255, 255))
+        hsv_lower, hsv_upper = self.colour_bounds.get(
+            self.colour, default_colour_bounds
+        )
+        imrange = cv2.inRange(
+            image,
+            numpy.array(hsv_lower),
+            numpy.array(hsv_upper)
+        )
+        if not self.menu:
+            frame = pygame.surfarray.make_surface(cv2.flip(imrange, 1))
+            screen.blit(frame, (100, 0))
+            pygame.display.update()
+        # Find the contours
+        contourimage, contours, hierarchy = cv2.findContours(
+            imrange, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE
+        )
+
+        # Go through each contour
+        found_area = -1
+        found_x = self.CROP_WIDTH+1
+        found_y = -1
+        biggest_contour = None
+        for contour in contours:
+            x, y, w, h = cv2.boundingRect(contour)
+            cx = x + (w / 2)
+            cy = y + (h / 2)
+            area = w * h
+            aspect_ratio = float(h)/w
+            if found_x > cx and area > self.MIN_CONTOUR_AREA and area < self.MAX_TARGET_SIZE and w < self.MAX_TARGET_WIDTH:
+                found_area = area
+                found_x = cx
+                found_y = cy
+                biggest_contour = contour
+        if found_area > self.MIN_CONTOUR_AREA:
```

Comment:
is this ever not true? surely the if statement on line 119 above ensures that an `area` is never assigned to `found_area` unless it is `area > self.MIN_CONTOUR_AREA`?

Reply:
this is really check if there were any contours found, but not in a readable way

Reply:
perhaps a comment? Be nice to Future You ;-)